### PR TITLE
feat: DataTable Pagination

### PR DESCRIPTION
**Pull request by unikvozm**
### Summary
#2416

- [x] add navigation to first and last page
- [x] fix bug with text displaying the last row
- [x] add select for number of rows

### Test plan

Screenshot 2021-01-10 at 17 51 57
Screenshot 2021-01-10 at 17 52 08
Screenshot 2021-01-10 at 17 52 08
Screenshot 2021-01-18 at 10 19 57

### CODE REVIEW

**Comment by Trancever** [comment]
Hey @unikvozm Is this PR ready for review? Could you please make a nice looking screenshot of the component and update the one we currently display in the documentation for DataTable.Pagination component?

**Comment by unikvozm** [comment]
> Hey @unikvozm Is this PR ready for review? Could you please make a nice looking screenshot of the component and update the one we currently display in the documentation for DataTable.Pagination component?

Hi @Trancever, unfortunately, I haven't got enough time to finish it =( only 1/3 of this is finished. Will try to get it ready this week.

**Comment by Trancever** [comment]
@unikvozm No worries! All good, take your time :) just wanted to check the status.

**Comment by unikvozm** [comment]
@Trancever, could you suggest me how to change the screenshot in the documentation, please?

**Comment by Angelk90** [comment]
@unikvozm , @Trancever : Are there any news?

**Comment by Angelk90** [comment]
@Trancever : Are there any news?

File: example/src/Examples/DataTableExample.tsx [hunk 1]
```diff
@@ -93,9 +98,14 @@ const DataTableExample = () => {
 
           <DataTable.Pagination
             page={page}
-            numberOfPages={Math.round(sortedItems.length / itemsPerPage)}
+            numberOfPages={Math.ceil(sortedItems.length / itemsPerPage)}
             onPageChange={(page) => setPage(page)}
             label={`${from + 1}-${to} of ${sortedItems.length}`}
+            optionsPerPage={optionsPerPage}
+            itemsPerPage={itemsPerPage}
+            setItemsPerPage={setItemsPerPage}
```
Comment (satya164) [inline example/src/Examples/DataTableExample.tsx:106]: Change this prop to `onItemsPerPageChange` or something like that

File: example/src/Examples/DataTableExample.tsx [hunk 2]
```diff
@@ -93,9 +98,14 @@ const DataTableExample = () => {
...
+            showFastPagination
```
Comment (satya164) [inline example/src/Examples/DataTableExample.tsx:107]: It's not clear from the prop name on what this does

**Review by satya164 — CHANGES_REQUESTED** [review]

**Comment by Angelk90** [comment]
@unikvozm , @Trancever : Are there any news?

**Comment by lukewalczak** [comment]
***UPDATES***

1. We've decided to not hide the pagination dropdown on smaller device, instead we're wrapping them into two lines.

2. Props are renamed, most of them are new, however existing one `label` was renamed to `selectPageDropdownLabel`, so it should be consider as a breaking change.

***SCREENSHOTS***

<details>
<summary>iPhone 8</summary>

portrait | landscape
--- | ---
Zrzut ekranu 2021-04-20 o 09 44 57 | Zrzut ekranu 2021-04-20 o 09 45 01

</details>

<details>
<summary>iPad</summary>

Zrzut ekranu 2021-04-20 o 09 55 21

</details>

<details>
<summary>Android</summary>

portrait | landscape
--- | ---
Zrzut ekranu 2021-04-20 o 09 47 40 | Zrzut ekranu 2021-04-20 o 09 47 49


</details>

<details>
<summary>web</summary>

Zrzut ekranu 2021-04-20 o 09 45 37

</details>

File: example/src/Examples/DataTableExample.tsx (hunk 1 above)
Comment (lukewalczak) [reply example/src/Examples/DataTableExample.tsx]: Done.

File: example/src/Examples/DataTableExample.tsx (hunk 2 above)
Comment (lukewalczak) [reply example/src/Examples/DataTableExample.tsx]: Renamed.

**Comment by Angelk90** [comment]
@unikvozm , @Trancever, @lukewalczak : Are there any news?

**Comment by brunohkbx** [comment]
Sure, will do!

File: src/components/DataTable/DataTablePagination.tsx [hunk 3]
```diff
@@ -91,46 +131,152 @@ type Props = React.ComponentPropsWithRef<typeof View> & {
...
+              </Text>
+              <Menu
+                visible={showSelect}
+                onDismiss={() => toggleSelect(!showSelect)}
+                anchor={
+                  <Button
+                    mode="outlined"
+                    onPress={() => toggleSelect(true)}
+                    style={styles.button}
+                    icon={'menu-down'}
```
Comment (dsznajder) [inline src/components/DataTable/DataTablePagination.tsx:183]:
```suggestion
                    icon="menu-down"
```

File: src/components/DataTable/DataTable.tsx [hunk 4]
```diff
@@ -39,36 +39,50 @@ type Props = React.ComponentPropsWithRef<typeof View> & {
...
  * import { DataTable } from 'react-native-paper';
  *
- * const MyComponent = () => (
- *   <DataTable>
- *     <DataTable.Header>
- *       <DataTable.Title>Dessert</DataTable.Title>
- *       <DataTable.Title numeric>Calories</DataTable.Title>
- *       <DataTable.Title numeric>Fat</DataTable.Title>
- *     </DataTable.Header>
+ * const optionsPerPage = [2, 3, 4];
```
Comment (brunohkbx) [inline src/components/DataTable/DataTable.tsx:42]: hey @lukewalczak can we also update this img `` ?. It looks so small on desktop
Screenshot from 2021-05-24 09-08-30

File: src/components/DataTable/DataTablePagination.tsx [hunk 5]
```diff
@@ -91,46 +131,152 @@ type Props = React.ComponentPropsWithRef<typeof View> & {
...
+  selectPageDropdownAccessibilityLabel,
   ...rest
 }: Props) => {
+  const [showSelect, toggleSelect] = React.useState(false);
   const labelColor = color(theme.colors.text).alpha(0.6).rgb().string();
 
   return (
-    <View {...rest} style={[styles.container, style]}>
-      <Text style={[styles.label, { color: labelColor }]} numberOfLines={1}>
+    <View
```
Comment (brunohkbx) [inline src/components/DataTable/DataTablePagination.tsx:152]: Accessibility wise I found this issue on android: 
bug

It focus on the whole card when I tried to focus on the row `Rows per page`

File: src/components/DataTable/DataTablePagination.tsx [hunk 6]
```diff
@@ -91,46 +131,152 @@ type Props = React.ComponentPropsWithRef<typeof View> & {
...
+      {...rest}
+      style={[styles.container, style]}
+      accessibilityLabel="pagination-container"
+    >
+      {numberOfItemsPerPageList &&
+        numberOfItemsPerPage &&
+        onItemsPerPageChange && (
+          <View style={styles.nonIconsContainer}>
+            <View
```
Comment (brunohkbx) [inline src/components/DataTable/DataTablePagination.tsx:161]: I think you can have just a single `View` here moving the style `style={styles.nonIconsContainer}` to this one

File: src/components/DataTable/DataTablePagination.tsx (hunk 5 above)
Comment (lukewalczak) [reply src/components/DataTable/DataTablePagination.tsx:152]: I think it's not related to DataTable component itself. I guess `Card` is blocking the access to the nested components in terms of a11y in the example - passing `<Card accessible={false} ...` can solve the issue.

File: src/components/DataTable/DataTable.tsx (hunk 4 above)
Comment (lukewalczak) [reply src/components/DataTable/DataTable.tsx:42]: Sure

File: src/components/DataTable/DataTablePagination.tsx (hunk 6 above)
Comment (lukewalczak) [reply src/components/DataTable/DataTablePagination.tsx:161]: Correct. Looks like I forgot to do that after moving `label` out of that `View`

**Review by dsznajder — APPROVED** [review]

**Review by brunohkbx — APPROVED** [review]
Looks good!

**Comment by sugata98** [comment]
The pagination control buttons remain black in dark mode, thus not visible. I am unable to fix this using theme prop. Any help?

**Comment by Angelk90** [comment]
@sugata98 : Try posting a screenshot and sample code so they will be more helpful in understanding what the problem is and how to fix it.

**Comment by sugata98** [comment]
`<DataTable.Pagination
                page={page}
                numberOfPages={Math.ceil(sortedItems.length / itemsPerPage)}
                onPageChange={page => setPage(page)}
                label={`${from + 1}-${to} of ${sortedItems.length}`}
                numberOfItemsPerPageList={numberOfItemsPerPageList}
                numberOfItemsPerPage={itemsPerPage}
                onItemsPerPageChange={onItemsPerPageChange}
                showFastPaginationControls
                selectPageDropdownLabel={'Rows per page'}
               theme={{
                  dark: theme.mode == 'light' ? false : true,
                  colors: {
                    text: theme.PRIMARY_TEXT_100,
                    primary: 'red',
                  },
                }}
              />`
BY using the theme prop I was able to change the text color to white in dark mode, but the dropdown and the pagination controls at the button remains white and dark respectively even if I change theme to dark.
WhatsApp Image 2021-08-11 at 2 17 30 PM

WhatsApp Image 2021-08-11 at 2 17 30 PM (1)

**Pull request merged**